ngi_reports/stockholm/project_summary.py

```diff
@@ -199,3 +229,55 @@ def get_order_dates(self):
         if self.creation_date:
             dates.append("_Report date:_ {}".format(self.creation_date))
         return ", ".join(dates)
+
+
+    def get_library_method(self):
+        """Get the library construction method and return as formatted string
+        """
+        if self.proj.get('application') == "Finished library":
+            return "Library was prepared by user."
+        try:
+            lib_head = ['Input', 'Type', 'Option', 'Category']
+            lib_meth = re.match(r'(.*) \[[0-9]+\]',self.proj_details['library_construction_method'])
+            if lib_meth:
+                return ("\n".join(["* {}: {}".format(lib_head[c],i.strip()) for c,i in enumerate(lib_meth.group(1).split(',')) \
```

Comment:
Yeah, at least I think that's how it works. This'd be way nicer in Perl though :wink: 
```perl
my lib_meth = (self.proj_details['library_construction_method']) =~ s/\[\d+\]$//;
```